src/main/groovy/org/pih/warehouse/inventory/CycleCountCustomItemCommand.groovy

```diff
@@ -0,0 +1,41 @@
+package org.pih.warehouse.inventory
+
+import grails.databinding.BindUsing
+import grails.validation.Validateable
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.product.Product
+import org.springframework.web.context.request.RequestContextHolder
+
+class CycleCountCustomItemCommand implements Validateable {
+
+    boolean recount
+
+    @BindUsing({ obj, source ->
+        Product product = Product.read(source['inventoryItem']['product'])
+        Date expirationDate = new Date(source['inventoryItem']['expirationDate'])
+        return InventoryItem.findByProductAndLotNumberAndExpirationDate(product, source['inventoryItem']['lotNumber'], expirationDate)
+    })
+    InventoryItem inventoryItem
```

Comment:
I don't know why I didn't think of this before, but looking at inventory item:
```
    static constraints = {
        product(nullable: false)
        lotNumber(nullable: true, unique: ['product'], maxSize: 255)
        expirationDate(shared:"expirationDateConstraint")

grails.gorm.default.constraints = {
    expirationDateConstraint(nullable:true, min: new Date("01/01/2000"))
}
```
so it looks like product and lot number is enough for uniqueness on inventory item. Maybe you don't even need to provide expirationDate as a part of the findBy here

Reply:
So I was wondering why we even have expirationDate as a field here if it's not being used anywhere. I can't remember what the original requirement was, but I wonder if the expectation is that if no inventoryItem exist for the given [product + lot number] pair, we should create it.

With your code currently, if the inventoryItem does not exist for the given [product + lot number] pair, it fails validation (because inventoryItem == null). This means expirationDate isn't used at all and we could remove it unless we're missing a requirement here to create the inventory item if it does not exist.